scripts/makefile_dump.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python3
+
+import subprocess
+from argparse import ArgumentParser
+from pathlib import Path
+from typing import Tuple, List, Dict
+from tempfile import NamedTemporaryFile
+import json
+
+
+LISTPARAMS_MAKEFILE = """
+listparams:
+\t@echo Start dumping params
+\t@echo APP_LOAD_PARAMS=$(APP_LOAD_PARAMS)
+\t@echo GLYPH_FILES=$(GLYPH_FILES)
+\t@echo ICONNAME=$(ICONNAME)
+\t@echo TARGET=$(TARGET)
+\t@echo TARGET_NAME=$(TARGET_NAME)
+\t@echo TARGET_ID=$(TARGET_ID)
+\t@echo APPNAME=$(APPNAME)
+\t@echo APPVERSION=$(APPVERSION)
+\t@echo API_LEVEL=$(API_LEVEL)
+\t@echo SDK_NAME=$(SDK_NAME)
+\t@echo SDK_VERSION=$(SDK_VERSION)
+\t@echo SDK_HASH=$(SDK_HASH)
+\t@echo Stop dumping params
+"""
+
+
+def run_cmd(cmd: str,
+            cwd: Path,
+            print_output: bool = False,
+            no_throw: bool = False) -> str:
+    print(f"[run_cmd] Running: {cmd} from {cwd}")
+
+    ret = subprocess.run(cmd,
+                         shell=True,
+                         stdout=subprocess.PIPE,
+                         stderr=subprocess.STDOUT,
+                         universal_newlines=True,
+                         cwd=cwd)
+    if no_throw is False and ret.returncode:
+        print(f"[run_cmd] Error {ret.returncode} raised while running cmd: {cmd}")
+        print("[run_cmd] Output was:")
+        print(ret.stdout)
+        raise ValueError()
+
+    if print_output:
+        print(f"[run_cmd] Output:\n{ret.stdout}")
+
+    return ret.stdout.strip()
+
+
+def get_app_listvariants(app_build_path: Path) -> Tuple[str, List[str]]:
+    # Using listvariants Makefile target
+    listvariants = run_cmd("make listvariants", cwd=app_build_path)
+    if "VARIANTS" not in listvariants:
+        raise ValueError(f"Invalid variants retrieved: {listvariants}")
+
+    # Drop Makefile logs previous to listvariants output
+    listvariants = listvariants.split("VARIANTS ")[1]
+
+    variants = listvariants.split(" ")
+    variant_param_name = variants.pop(0)
+    return variant_param_name, variants
```

Comment:
I have seen many apps not defining correctly the `listvariants` target in their makefile.
We need to add something here like:
```suggestion
    variants = listvariants.split(" ")
    variant_param_name = variants.pop(0)
    assert variants, "At least one variant should be defined in the app Makefile"
    return variant_param_name, variants
```

Reply:
Done